navpat.py

```diff
@@ -41,9 +44,18 @@ def min_dist(data, infinity=None):
 
 prev_cones = []
 prev_near = False
+last_laser_update_time = None
+
 def detect_near_extension(robot, id, data):
-    global prev_near
+    global prev_near, last_laser_update_time
```

Comment:
Can you, in the future, consider removing this pattern of global variables? AFAIK you can register member function as a callback which is bound to an instance that can hold your state. No need for global variables.

Reply:
OK, thanks - good point. Maybe we should revise extensions - there could be some initialization call during registration and then regular update?? An alternative is your suggestion - I think that I used registered member functions in Eduro project.

Reply:
There is a new commit with refactoring (LaserDetector class is used instead of global variables).
Is it closer to your idea? Thanks